File: Eglantine/Data/Events/MainMenu_events.lua
--Load the relevent assemblies and types
luanet.load_assembly "Eglantine"
EVENT_MANAGER = luanet.import_type "Eglantine.Engine.EventManager"

--Functions related to .NET classes and objects

--Sets the lua global Event to the current EventManager.Instance--
function loadEventManager()
	print("EventManager loaded.")
	Event = EVENT_MANAGER.Instance
end

loadEventManager()


menuPhase = 0


function showSplashScreen()
	runCoroutine(function()
		Event:PlaySong("Toxic Night", .7)
		waitSeconds(2)
		Event:MainMenuFadeIn("splash", 3)
		waitSeconds(7)
		Event:MainMenuFadeOut("splash", 3)
		waitSeconds(3)
		Event:NextMenuPhase()
	end)
end


function titleFadeIn()
	runCoroutine(function()
		Event:MainMenuHideElement("splash")
		
		Event:MainMenuFadeIn("background", 5);
		waitSeconds(6)
		Event:MainMenuFadeIn("title", 3);
		waitSeconds(3)
		Event:NextMenuPhase()
	end)
end


function mainMenu()
	Event:MainMenuFadeIn("background", 0);
	Event:MainMenuFadeIn("title", 0);

	Event:ShowMainMenu()
end

function onStartNewGame()
	runCoroutine(function()
		Event:PlaySound("windowbreak")
		Event:FadeMusic(0,0)
		Event:LockMenuInput()
		waitSeconds(2.5)
		Event:MainMenuFadeOut("background", 3)
		Event:MainMenuFadeOut("title", 3)
		waitSeconds(3.5)
		Event:PlayStorySequence("openingCutscene")
	end)
end

menuEvents = {
	[1] = showSplashScreen,
	[2] = titleFadeIn,
	[3] = mainMenu
}

File: Eglantine/Eglantine/Eglantine/Scripts/MainMenuEvents.cs
using System;
using System.Collections.Generic;
using ObjectivelyRadical.Scheduler;

namespace Eglantine.Engine
{
	public static partial class GameEvents
	{
		public static int menuPhase = 0;
		public static Script[] menuEvents = new Script[4]
		{
			null,
			showSplashScreen,
			titleFadeIn,
			mainMenu
		};

		public static IEnumerator<ScriptPauser> showSplashScreen()
		{
			EventManager.Instance.PlaySong("Toxic Night", .7f);
			yield return waitSeconds(2);
			EventManager.Instance.MainMenuFadeIn("splash", 3);
			yield return waitSeconds(7);
			EventManager.Instance.MainMenuFadeOut("splash", 3);
			yield return waitSeconds(3);
			EventManager.Instance.NextMenuPhase();
		}

		public static IEnumerator<ScriptPauser> titleFadeIn()
		{
			EventManager.Instance.MainMenuHideElement("splash");
			EventManager.Instance.MainMenuFadeIn("background", 5);
			yield return waitSeconds(6);
			EventManager.Instance.MainMenuFadeIn("title", 3);
			yield return waitSeconds(3);
			EventManager.Instance.NextMenuPhase();
		}

		public static IEnumerator<ScriptPauser> mainMenu()
		{
			EventManager.Instance.MainMenuFadeIn("background", 0);
			EventManager.Instance.MainMenuFadeIn("title", 0);
			EventManager.Instance.ShowMainMenu();

			yield return null;
		}

		public static IEnumerator<ScriptPauser> onStartNewGame()
		{
			EventManager.Instance.PlaySound("windowbreak");
			EventManager.Instance.StopMusic();
			EventManager.Instance.LockMenuInput();
			yield return waitSeconds(2.5f);

			EventManager.Instance.MainMenuFadeOut("background", 3);
			EventManager.Instance.MainMenuFadeOut("title", 3);
			yield return waitSeconds(3.5f);

			EventManager.Instance.PlayStorySequence("openingCutscene");
		}
	}
}

